### ISSUE TITLE
.flac support

### ISSUE THREAD

**Issue by KeitIG** [KeitIG/museeks#53]
It is said here it's supposed to work, but it doesn't https://www.chromium.org/audio-video

_edit/tl;dr: 21/11/2016: Flac codecs shipped with Chromium 56 (January 31st, 2017), Electron may follow a few weeks later_

**Comment by KeitIG** [comment]
blocked by https://code.google.com/p/chromium/issues/detail?id=93887

**Comment by ploctaux** [comment]
Check again https://www.chromium.org/audio-video, flac is now added :)

**Comment by KeitIG** [comment]
Great, Museeks will support Flac as soon as Electron will be based on a the correct chromium version :)

**Comment by ploctaux** [comment]
Super :)

--
Philippe Loctaux <phil@philippeloctaux.com>

**Comment by SamHH** [comment]
What's the deal with this - is it supported in Chromium yet or not?

**Comment by KeitIG** [comment]
Hey @SamHH ! Unfortunately, the FLAC codec is only available in the Chrome OS build (for reasons...).

A lot of developers are trying to bump up this issue, but still no answer from the Chrome team :/

**Comment by YurySolovyov** [comment]
how about https://github.com/audiocogs/aurora.js ?

**Comment by KeitIG** [comment]
Well, I know this is a killer feature, aurora.js could be a solution, but the current implementation with the Web audio API is really simple and easy to deal with.

I'll try to investigate workaround/refactorings. But this is not a priority: I'd like to finish playlists first.

**Comment by KeitIG** [comment]
Adding that issue to 1.0 milestone, hope that things will have moved a bit when we'll get there. Otherwise I will have to implement a new way to play audio.

**Comment by KeitIG** [comment]
Things are moving.

- https://bugs.chromium.org/p/chromium/issues/detail?id=93887#c45
- https://bugs.chromium.org/p/chromium/issues/detail?id=642219

**Comment by KeitIG** [comment]
Shipped in Canary 
https://bugs.chromium.org/p/chromium/issues/detail?id=93887#c70

**Comment by MrBlenny** [comment]
Any updates?

**Comment by KeitIG** [comment]
@MrBlenny We just have to wait an Electron release including Chrome 56. I'm not really as confident as before as the beta version of electron includes Chrome... 54 :sob: 

So I can't give you any ETA.

**Comment by MrBlenny** [comment]
Out of interest, do you know how flac support is achieved [here](https://github.com/mashaal/destroyer)?

**Comment by YurySolovyov** [comment]
@MrBlenny with
`"fluent-ffmpeg": "2.0.1",`

**Comment by MrBlenny** [comment]
I suspected as much. 
Is implementation of a `ffmepg` solution difficult or just a time cost you are trying to avoid given the potential release of 56 soon? 

Given that it took 4 months for electron to upgrade from chrome 53 to 54 - it seems we could be waiting quite a while longer for 54 to move to 56.

**Comment by KeitIG** [comment]
@MrBlenny FFMPEG is something we will most probably support in the future. @YurySolovyov has already done an excellent job with it, there are just a few drawbacks we still need to discuss about, like getting metadatas in a as robust way as `musicmetadata`, the interface between Museeks and FFMPEG...

**Comment by YurySolovyov** [comment]
@MrBlenny ffmpeg will require significant changes, so the way we could make it work is by shipping flac from Chrome first, then we'll implement ffmpeg based approach.

**Comment by vprigent** [comment]
Chrome 56 is on it's way with electron https://github.com/electron/electron/pull/8501

**Comment by levrik** [comment]
@MrBlenny The upgrade from Chrome 53 to 54 took so long because the Chrome team switched their build system from gyp to gn. From what I noticed, it was really difficult to make that work with Electron.

**Comment by KeitIG** [comment]
Finally we're good. Fixed by #322
